gnowsys-ndf/gnowsys_ndf/ndf/templatetags/ndf_tags.py

```diff
@@ -1206,10 +1206,10 @@ def get_prior_node(node_id):
 	if topic_GST._id in obj.member_of:
 
 		if obj.prior_node:
-			for each in obj.prior_node:
-				node = node_collection.one({'_id': ObjectId(each) })
-				prior.append(( node._id , node.name ))
-
+	#	for each in obj.prior_node:
+#			node = node_collection.one({'_id': ObjectId(each) })
+	#		prior.append(( node._id , node.name ))
+			prior=[(node_collection.one({'_id': ObjectId(each) })._id,node_collection.one({'_id': ObjectId(each) }).name) for each in obj.prior_node]
```

Comment:
We do need to reduce the no. of DB queries as much as possible. In above (modified/updated) code you have  optimized code with one line `for` loop but, you are firing two queries instead of one.

So, possibly optimize the code with minimal-no./existing-no. of DB queries.